server/service/post-service.ts

```diff
@@ -0,0 +1,48 @@
+import { Dependencies } from '../types';
+import { Repository } from 'typeorm';
+import { Post } from '../entities/post';
+import { User } from '../entities/user';
+import { plainToClass } from 'class-transformer';
+import { QueryDeepPartialEntity } from 'typeorm/query-builder/QueryPartialEntity';
+import { UserService } from './user-service';
+// Just a middleman currently?
+export class PostService {
+  repository: Repository<Post>;
+
+  constructor(options: Dependencies) {
+    this.repository = options.postRepository;
+  }
+
+  find() {
+    return this.repository.find();
+  }
+
+  findOne(id: number) {
+    return this.repository.findOne(id);
+  }
+
+  // Should this get access to service from upperlayer or have access to user repo on its own?
+  async save(ownerId: number, category: string, message: string, userService: UserService) {
+    try {
+      const temp = await userService.findOne(ownerId);
+      const user = plainToClass(User, temp);
+      if (user) {
+        const post: Post = new Post(user, category, message);
+        return await this.repository.save(post);
+      } else {
+        return undefined;
+      }
+    } catch (error) {
+      return undefined;
+    }
+  }
+
+  delete(id: number) {
+    return this.repository.delete(id);
+  }
+
+  update(id: number, obj: QueryDeepPartialEntity<Post>) {
+    this.repository.update(id, obj);
```

Comment:
Hmm isn't this async operation that we have to wait before returning? Also can you check what `update` returns? Could it be the updated object?

Reply:
This we went through in tv and I checked the updateResult object which was just what I expected! Fixed all the update methods in to using .then though!